src/app/features/search/pages/search-detail-page/search-detail-page.component.ts

```diff
@@ -0,0 +1,236 @@
+import { Location, NgClass, NgFor, NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, DestroyRef, inject, signal, effect } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AlertService } from '@core/services/alert/alert.service';
+import { CarriageService } from '@core/services/carriage/carriage.service';
+import { CarriagePreviewComponent } from '@features/admin/components/carriage-preview/carriage-preview.component';
+import { RoadSection, Trip } from '@features/search/interfaces/search-route-response.interface';
+import { SearchApiService } from '@features/search/services/search-api/search-api.service';
+import { TuiChip, TuiTab, TuiTabs } from '@taiga-ui/kit';
+import { map, switchMap } from 'rxjs';
+import { TuiCurrencyPipe } from '@taiga-ui/addon-commerce';
+import {
+  BookSeats,
+  CarriageList,
+  FreeSeat,
+  Price,
+  TripService,
+  SelectedOrder,
+} from '@features/search/services/trip/trip.service';
+import { OrderPanelComponent } from '@features/search/components/order-panel/order-panel.component';
+import { OrdersApiService } from '@features/orders/services/orders-api/orders-api.service';
+import { TuiButton } from '@taiga-ui/core';
+import { OrderRequest } from '@features/search/interfaces/order.request.interface';
+
+@Component({
+  selector: 'dd-search-detail-page',
+  standalone: true,
+  imports: [
+    TuiTabs,
+    TuiTab,
+    NgFor,
+    NgClass,
+    NgIf,
+    CarriagePreviewComponent,
+    TuiCurrencyPipe,
+    OrderPanelComponent,
+    TuiChip,
+    TuiButton,
+  ],
+  templateUrl: './search-detail-page.component.html',
+  styleUrl: './search-detail-page.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchDetailPageComponent {
+  private readonly route = inject(ActivatedRoute);
+
+  private readonly router = inject(Router);
+
+  private readonly location = inject(Location);
+
+  private readonly searchApiService = inject(SearchApiService);
+
+  private readonly ordersApiService = inject(OrdersApiService);
+
+  private readonly carriageService = inject(CarriageService);
+
+  private readonly tripService = inject(TripService);
+
+  private readonly destroy = inject(DestroyRef);
+
+  private readonly alert = inject(AlertService);
+
+  public carriages = this.carriageService.carriages;
+
+  private readonly _ride = signal<Trip | null>(null);
+
+  public readonly ride = this._ride.asReadonly();
+
+  private readonly _isBookSeat = signal(false);
+
+  public readonly isBookSeat = this._isBookSeat.asReadonly();
+
+  public orderId = signal(0);
+
+  public options = signal({ isClick: true, isShowTitle: false });
+
+  public fromStation!: number;
+
+  public toStation!: number;
```

Comment:
Is it really okay to use non-null assertion operator here? Is it possible that fromStation or toStation could be null??

Reply:
Okay I see the point now